### PR TITLE
MM-16425 Fixes #12614 : Show single image thumbnails in compact view

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
In edit mode, for image upload post
- [x] Styling: Remove the bolding on the image name
- [x] Placement: Image name appears on a new line with expand/Collapse control
- [x] Image thumbnail size: 175px (50%) of 350px 

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12614
Jira https://mattermost.atlassian.net/browse/MM-16425

#### Screenshots
In compact display setting :
Screenshot from 2019-12-03 16-26-19

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@hanzei should i include an e2e test ?

**Comment by hanzei** [comment]
e2e are always welcome and much appreciated

**Comment by M-ZubairAhmed** [comment]
@hanzei can you please point to which file in e2e should i include the test

**Comment by hanzei** [comment]
@srkgupta Can you help with this question?

**Comment by srkgupta** [comment]
@M-ZubairAhmed 
Thanks for offering to add an E2E test. You can add a E2E test within messaging directory. You can create a new test file for this.

**Comment by hanzei** [comment]
@adamjclarkson You can set one up using the `Setup Cloud Test Server` label

**Comment by M-ZubairAhmed** [comment]
This is a good question @adamjclarkson, i also would like to hear what andrew says, the max height for non compact was also fixed to 350px before this pr,

**Comment by andrewbrown00** [comment]
@M-ZubairAhmed @adamjclarkson thanks for the feedback on this! I updated the ticket/Jira to reflect the 175px dimension. DP is used in for Android devices and was an oversight from copying and pasting specs from the Mobile Image Preview updates I made while working on this ticket.

**Comment by cpanato** [comment]
will move this test server to another cluster.

**Comment by cpanato** [comment]
will terminate this test server, if you need one please set the label again

**Comment by M-ZubairAhmed** [comment]
Has there been any update om this one ?

**Comment by jgilliam17** [comment]
@deanwhillier @adamjclarkson 
Can you please help move this PR along by adding Dev and UX reviewers. Thank you.

File: e2e/cypress/support/ui_commands.js [hunk 1]
```diff
@@ -86,7 +86,6 @@ Cypress.Commands.add('toAccountSettingsModalChannelSwitcher', (username, setToOn
 /**
  * Change the message display setting
  * @param {String} setting - as 'STANDARD' or 'COMPACT'
- * @param {String} username - User to login as
```
Comment (devinbinnie) [inline e2e/cypress/support/ui_commands.js]: How come this was removed?

Comment (M-ZubairAhmed) [reply e2e/cypress/support/ui_commands.js]: username is not used in the function

File: components/single_image_view/single_image_view.jsx [hunk 2]
```diff
@@ -107,8 +109,14 @@ export default class SingleImageView extends React.PureComponent {
...
         }
 
+        // Add compact display class to image class if in compact mode
+        if (compactDisplay) {
+            minPreviewClass += 'compact-display';
+        }
+
         const toggle = (
             <button
+                id={`single_file_upload__toggle_${this.props.postId}`}
```
[the remark this answers is not in the document]
Comment (M-ZubairAhmed) [reply components/single_image_view/single_image_view.jsx]: i have got away with ids altogether, and moved to much more user intuitive way of testing, as this PR is pretty old i wasn't aware of `Testing-Library` methodology. Now tests are more in tune with that

**Comment by hmhealey** [comment]
Should the bolding be removed when you're not in compact mode? I'm seeing it even without compact mode enabled.
Screen Shot 2020-01-08 at 4 44 23 PM

**Comment by hmhealey** [comment]
^ @asaadmahmood For that question

**Comment by hmhealey** [comment]
The code for this looks good otherwise

**Comment by M-ZubairAhmed** [comment]
@hmhealey @assadmahmood let me know what changes needs to be made next

**Comment by andrewbrown00** [comment]
> Should the bolding be removed when you're not in compact mode? I'm seeing it even without compact mode enabled.
> Screen Shot 2020-01-08 at 4 44 23 PM

The **bolding** should not be removed in standard view. It was decided to remove the bolding from compact view only to keep the attention on the username since on compact view we remove the avatar.

**Comment by M-ZubairAhmed** [comment]
@andrewbrown00 Hi, so as per your comment, the current version is ok ?

**Comment by andrewbrown00** [comment]
> @andrewbrown00 Hi, so as per your comment, the current version is ok ?

Hi @M-ZubairAhmed did the bolding get removed for the standard view in this PR or a different one. Will you confirm it's still bold for standard view.

The current version is good in compact view
image

**Comment by M-ZubairAhmed** [comment]
@andrewbrown00 yes previously bolding was removed for both compact and standard. I have made new changes where standard retains bolding and compact removes bolding for image name.

**Comment by cpanato** [comment]
removin the tests server for now, if you need that again, please label again.

**Comment by adamjclarkson** [comment]
@hmhealey can we get a review on this today?

**Pull request merged**